packages/eslint-plugin-svelte/src/utils/svelte-context.ts

```diff
@@ -0,0 +1,259 @@
+import type { RuleContext } from '../types.js';
+import fs from 'fs';
+import path from 'path';
+import { getPackageJsons } from './get-package-json.js';
+import { getFilename, getSourceCode } from './compat.js';
+
+const isRunInBrowser = !fs.readFileSync;
+
+export type SvelteContext = (
+	| ({
+			svelteVersion: '3/4';
+	  } & {
+			svelteFileType: '.svelte' | null;
+			runes: null;
```

Comment:
Why `null` here and not `false`? This conflates it with a non-svelte file, rather than a Svelte 5 file in non-runes mode, which I think is semantically much closer..

Reply:
Because Svelte 3/4 doesn't have runes. `false` means "There is a concept of runes, but it is not applied in this file" but if that concept does not exist, I uses `null`.
Anyway we can recognize this via typing.